### PR TITLE
usm: Added go-tls documentation

### PR DESCRIPTION
**Pull request by guyarb**
### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/universal_service_monitoring/setup.md [hunk 1]
```diff
@@ -760,6 +760,51 @@ DD_SYSTEM_PROBE_PROCESS_SERVICE_INFERENCE_ENABLED=true
 
 {{< /tabs >}}
 
+### Go TLS Monitoring
+
+<div class="alert alert-info">
+Universal Service Monitoring is available in <strong>beta</strong> to monitor TLS encrypted traffic from services implemented in Golang.
+
+<i>Note: Go HTTPs servers can upgrade HTTP1.1 protocol to HTTP/2 which is supported in private beta, please reach to your account manager for further details<i>
```
Comment (sunhay) [inline content/en/universal_service_monitoring/setup.md:768]: The text is awkwardly sized here -- I think it's fine to move this note out of this alert and alongside the agent version requirement.

Screenshot 2024-03-07 at 7 28 41 AM

**Comment by sunhay** [comment]
It's probably worth also updating https://docs-staging.datadoghq.com/guy.arbitman/usm-go-tls-public-beta/universal_service_monitoring/additional_protocols/ now that we've moved Istio and Go TLS into public beta.

@conward2121 do you have a preference on how we're organizing all this content?

**Comment by guyarb** [comment]
@sunhay addressed the suggestions

**Review by sunhay — APPROVED** [review]

File: content/en/universal_service_monitoring/additional_protocols.md [hunk 2]
```diff
@@ -30,18 +30,11 @@ Universal Service Monitoring can discover the following third-party API endpoint
 Request access and ensure your Datadog Agent is upgraded to at least version 7.49.0 to gain access to the following features.
 
 <div class="alert alert-info"><p>The following support does <strong>not</strong> include:
-<ul><li>Monitoring services that use TLS with gRPC/HTTP2</li> 
-<li>Monitoring Go-TLS applications where the Golang server upgrades the HTTP connection to HTTP2.</li></ul></p></div>
+<ul><li>Monitoring services that use gRPC/HTTP2</li></ul></p></div>
```
[the remark this answers is not in the document]
Comment (guyarb) [reply content/en/universal_service_monitoring/additional_protocols.md:33]: removed completely. thanks for raising that

**Review by brett0000FF — APPROVED** [review]
Looks good--thanks! 👍

**Pull request merged**